.github/workflows/teamcity-services-diff-check-weekly.yml

```diff
@@ -11,57 +11,60 @@ on:
-      - name: Checkout Repository
-        uses: actions/checkout@v4
-
-      - name: Setup Go
-        uses: actions/setup-go@v4
-        with:
-          go-version: '^1.20'
-
-      - name: Download built artifacts - GA provider
-        uses: actions/download-artifact@v2
-        with:
-          name: artifact-terraform-provider-google
-          path: artifacts
-
-      - name: Unzip the artifacts and delete the zip
-        run: |
-          unzip -o artifacts/output.zip -d ./provider
-          rm artifacts/output.zip
-
-      - name: Download built artifacts - Beta provider
-        uses: actions/download-artifact@v2
-        with:
-          name: artifact-terraform-provider-google-beta
-          path: artifacts
-
-      - name: Unzip the artifacts and delete the zip
-        run: |
-          unzip -o artifacts/output.zip -d ./provider
-          rm artifacts/output.zip
-
-      - name: Check that new services have been added to the TeamCity configuration code
-        run: |
-          # Create lists of service packages in providers
-          ls provider/google/services > tools/teamcity-diff-check/services_ga.txt
-          ls provider/google-beta/services > tools/teamcity-diff-check/services_beta.txt
-
-          # Run tool to compare service packages in the providers vs those listed in TeamCity config files
-          cd tools/teamcity-diff-check
-          go run main.go -service_file=services_ga
-          go run main.go -service_file=services_beta
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      runs-on: ubuntu-22.04
+      steps:
+        - name: TeamCity Google Provider Generate
+          uses: ./.github/actions/build-downstream
+          with:
+            repo: 'terraform-provider-google'
+            token: '$GITHUB_TOKEN'
+    
+        - name: TeamCity Google Beta Provider Generate
+          uses: ./.github/actions/build-downstream
+          with:
+            repo: 'terraform-provider-google-beta'
+            token: '$GITHUB_TOKEN'
+    
+        - name: Checkout Repository
+          uses: actions/checkout@v4
+    
+        - name: Setup Go
+          uses: actions/setup-go@v4
+          with:
+            go-version: '^1.20'
+    
+        - name: Download built artifacts - GA provider
+          uses: actions/download-artifact@v2
+          with:
+            name: artifact-terraform-provider-google
+            path: artifacts
+    
+        - name: Unzip the artifacts and delete the zip
+          run: |
+              unzip -o artifacts/output.zip -d ./provider
+              rm artifacts/output.zip
+    
+        - name: Download built artifacts - Beta provider
+          uses: actions/download-artifact@v2
+          with:
+            name: artifact-terraform-provider-google-beta
+            path: artifacts
+    
+        - name: Unzip the artifacts and delete the zip
+          run: |
+              unzip -o artifacts/output.zip -d ./provider
+              rm artifacts/output.zip
```

Comment:
I don't think you'd need to download artifacts (see other comment) as I believe the steps of the `teamcity-services-diff-check` job would all run on the same runner? Wouldn't we be using the same machine to build the providers, upload them, download them, and then reference them in later steps?